src/Components/Server.java

```diff
@@ -0,0 +1,92 @@
+package Components;
+
+import java.io.*;
+import java.net.ServerSocket;
+import java.net.Socket;
+import DataModels.Operator;
+
+public class Server
+{
+    private ServerSocket serverSocket;
+    private Socket socket;
+    private DataOutputStream output;
+    private DataInputStream input;
+
+    private final String SEPARATOR = "<>";
+
+    public Server(int port)
+    {
+        try {
+            serverSocket = new ServerSocket(port);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void waitForClient()
+    {
+        try {
+            socket = serverSocket.accept();
+            output = new DataOutputStream(socket.getOutputStream());
+            input = new DataInputStream(new BufferedInputStream(socket.getInputStream()));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        waitForResponse();
+    }
+
+    private void waitForResponse()
+    {
+        String data = "";
+        while(true)
+        {
+            try {
+                data = input.readUTF();
+                if (data.equals("finish"))
+                {
+                    close();
+                    break;
+                }
+                output.writeUTF(calculate(data));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private String calculate(String data)
```

Comment:
you must extract operator and operands before calling this method(they must be arguments of the method)